providers/base/bin/virtualization.py

```diff
@@ -734,19 +734,20 @@ def __init__(self, template=None, image=None):
         self.default_remote = "ubuntu:"
         self.os_version = get_release_to_test()
 
-    def run_command(self, cmd):
+    def run_command(self, cmd, error_msgs=True):
         task = RunCommand(cmd)
         if task.returncode != 0:
             logging.error('Command {} returned a code of {}'.format(
                 task.cmd, task.returncode))
             logging.error(' STDOUT: {}'.format(task.stdout))
-            logging.error(' STDERR: {}'.format(task.stderr))
+            if error_msgs:
+                logging.error(' STDERR: {}'.format(task.stderr))
             return False
         else:
             logging.debug('Command {}:'.format(task.cmd))
             if task.stdout != '':
                 logging.debug(' STDOUT: {}'.format(task.stdout))
-            elif task.stderr != '':
+            elif task.stderr != '' and error_msgs:
```

Comment:
this `elif` is a bit weird, a task may have both `stdout and stderr`. I would change this to:
```python
if task.stdout:
  logging.debug(' STDOUT: {}'.format(task.stdout))
if task.stderr and error_msgs: # <- if you change the name, change it here as well
  logging.debug(' STDERR: {}'.format(task.stderr))
if not (task.stderr or task.stdout): 
  # we could be smart here and do elif not task.stdout
  # but I think this is a bit cleaner
  logging.debug(' Command returned no output')
```

Minor: `if string_var != ''` == `if string_var`